### DISCUSSION TITLE
Apples M1 CPU

### DISCUSSION THREAD

**Discussion by salbeik** [airsdk/Adobe-Runtime-Support#691] [General]
I apologies if this is a stupid question, there might be things I don't really understand. 
+++
As a developer my fear is of course at some point not to be able to have my as3 apps working on Apples new M1 CPU, especially when Apple at some point removes the emulator for running intel apps. I don't own one of the new Mac, so I can't test. But if I would run Inteliji IDEA + Harman air sdk on a new M1 CPU would it compile a application that natively runs on the new M1 CPU or would it create a "intel cpu app" that runs on the M1 Cpu through a emulator ? Obviously I'm hoping to be able to make a native app that doesnt require the build in emulator

**Comment by marchbold** [comment]
There is an active issue tracking development of the AIR SDK to be native on Apple Silicon https://github.com/Gamua/Adobe-Runtime-Support/issues/359

Currently though AIR applications will run through the rosetta 2 emulator, this doesn't require any changes and performs quite well.
